ujt/callbacks.py

```diff
@@ -91,53 +91,66 @@ def update_graph_elements(
 
     ctx = dash.callback_context
     triggered_id, triggered_prop, triggered_value = utils.ctx_triggered_info(ctx)
+
     print(triggered_id, triggered_prop, triggered_value)
+
     if triggered_id == "virtual-node-update-signal" and triggered_value != constants.OK_SIGNAL:
         # No-op if the validation signal isn't OK
         raise PreventUpdate
 
     node_name_message_map, client_name_message_map = state.get_message_maps()
     virtual_node_map = state.get_virtual_node_map()
 
-    if triggered_id == "refresh-sli-button":
-        state.clear_sli_cache(
-        )  # in future, conditionally clear this based on timestamp
-        sli_list = state.get_slis()
-        node_name_message_map = transformers.apply_slis_to_node_map(
-            sli_list,
-            node_name_message_map)
-
-    # Perform status computation.
-    compute_status.reset_node_statuses(node_name_message_map)
-    compute_status.reset_client_statuses(client_name_message_map)
-    compute_status.reset_node_statuses(virtual_node_map)
-
-    # combine the two maps of nodes into one dictionary
-    # use duck typing -- is this pythonic or a hack?
-    all_nodes_map = {**node_name_message_map, **virtual_node_map}  #type: ignore
-    compute_status.compute_statuses(
-        all_nodes_map,
-        client_name_message_map,
-    )
-
-    state.set_node_name_message_map(node_name_message_map)
-    state.set_client_name_message_map(client_name_message_map)
-    state.set_virtual_node_map(virtual_node_map)
-
-    # TODO: memoize this call per Ken's suggestion. This will be refactored when implementing server.
-    # something like: elements = state.get_cytoscape_elements()
-    elements = converters.cytoscape_elements_from_maps(
-        node_name_message_map,
-        client_name_message_map,
-    )
+    elements = state.get_cytoscape_elements()
+
+    # This condition determines if we need to recompute node statuses.
+    # We recompute node statuses on startup and when SLIs are refreshed
+    # The two following if statements can be combined later.
+    # Getting SLIs will return random values from the server,
```

Comment:
I would treat the client code as if the server were sending real values, it's a little weird to have this comment in here. In the future the app should support connecting to a real server or a test server, so the client should be agnostic to where the server is getting its data from.